azurerm/internal/services/compute/validation.go

```diff
@@ -103,3 +105,11 @@ func validateDiskSizeGB(v interface{}, _ string) (warnings []string, errors []er
 	}
 	return warnings, errors
 }
+
+func validateDedicatedHostGroupName() func(i interface{}, k string) (warnings []string, errors []error) {
+	return validation.StringMatch(regexp.MustCompile(`^[^_\W][\w-.]{0,78}[\w]$`), "")
```

Comment:
I suspect this regex may need tweaking, but since I'm unable to put my hands on the naming rules, I'm not completely sure. Worth checking later.